File: ios/Sources/AIDungeonMasterApp/SessionInfo.swift
import Foundation

/// Player identity returned by `POST /v2/session`.
public struct SessionInfo: Equatable, Sendable {
    public let sessionId: String
    public let token: String
    public let displayName: String
    public let expiresAtEpochSeconds: Int64
    public let createdAtEpochSeconds: Int64

    public init(
        sessionId: String,
        token: String,
        displayName: String,
        expiresAtEpochSeconds: Int64 = 0,
        createdAtEpochSeconds: Int64 = 0
    ) {
        self.sessionId = sessionId
        self.token = token
        self.displayName = displayName
        self.expiresAtEpochSeconds = expiresAtEpochSeconds
        self.createdAtEpochSeconds = createdAtEpochSeconds
    }

    /// Short id for chrome (first 8 chars).
    public var shortId: String {
        sessionId.count <= 8 ? sessionId : String(sessionId.prefix(8))
    }
}

File: android/app/src/main/java/com/xai/dungeonmaster/android/SessionInfo.kt
package com.xai.dungeonmaster.android

/**
 * Player identity returned by `POST /v2/session`.
 * Persisted via [SessionStore] so relaunches can restore the same world.
 */
data class SessionInfo(
    val sessionId: String,
    val token: String,
    val displayName: String,
    val expiresAtEpochSeconds: Long = 0L,
    val createdAtEpochSeconds: Long = 0L,
) {
    /** Short id for the UI chrome (first 8 chars). */
    fun shortId(): String = if (sessionId.length <= 8) sessionId else sessionId.take(8)

    fun isExpired(nowEpochSeconds: Long = System.currentTimeMillis() / 1000L): Boolean {
        if (expiresAtEpochSeconds <= 0L) return false
        // Small skew so we re-mint before the server rejects the JWT.
        return nowEpochSeconds >= (expiresAtEpochSeconds - 30)
    }

    /** Seconds until JWT expiry (0 if missing/expired). */
    fun secondsUntilExpiry(nowEpochSeconds: Long = System.currentTimeMillis() / 1000L): Long {
        if (expiresAtEpochSeconds <= 0L) return 0L
        return (expiresAtEpochSeconds - nowEpochSeconds).coerceAtLeast(0L)
    }
}
